### PR TITLE
[JENKINS-54052] Honor quietPeriod on children of multibranch pipeline job

### PR DESCRIPTION
**Pull request by g-doc**
Updated PR with change provided in comments of 

full credits to @atanasenko in PR #131

### CODE REVIEW

**Comment by bitwiseman** [comment]
@g-doc  This is great!  Still 3 test failures.  Can your reproduce them locally?

**Comment by bitwiseman** [comment]
Build was clean on rerun.  Trying again to see if we are stable.

**Comment by bdellegrazie** [comment]
Any chance this could be re-examined? The test failure looks like failed infra (channel closed).

**Comment by brianjmurrell** [comment]
This would be useful in GitHub if one has GitHub Actions that pushes a commit to one's PR -- to prevent the the PR from building prior to the GH Actions adding the new commit.

Can it be resuscitated?

**Comment by addib** [comment]
Now, that the build has been successful on three consecutive runs, are we good to merge?

File: src/test/java/integration/harness/BasicBranchProjectFactory.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ public BasicBranchProjectFactory() {
     public FreeStyleProject newInstance(Branch branch) {
         FreeStyleProject job = new FreeStyleProject(getOwner(), branch.getEncodedName());
         setBranch(job, branch);
+        try {
+            job.setQuietPeriod(0);
```
Comment (bitwiseman) [inline src/test/java/integration/harness/BasicBranchProjectFactory.java:51]: For test stability we maintain the previous quiet period of `0`.  But live behavior will change.

**Comment by bitwiseman** [comment]
I approve, but I'm thinking this through a bit deeper.  We should get some additional testing, both manual and running other plugin tests using this version.

**Comment by addib** [comment]
@bitwiseman, I have performed some quick manual tests with `branch-api-2.5.9-rc884.9435363784d2.hpi` available in the incremental repo with:
- Jenkins 2.235.2
- Pipeline: Multibranch 2.21
- GitHub Enterprise 2.20.7

The following are my observations:
- In declarative pipelines, `quietPeriod(<num>)` is respected for both PR creation and merge
- The specified quiet period takes effect immediately on the very first run. However, removing it takes effect only after the next run.
- The global quiet period is respected if not specified in Jenkinsfile. If specified in Jenkinsfile, this takes precedence over the global settings.
- PR builds triggered via PR comments (GitHub PR Comment Build Plugin 2.1) do not consider the quiet period. I guess this must be the expected behavior.
- The quiet period is not visible in multibranch project configuration GUI but is visible in the branch configuration GUI, though it remains uneditable

I can help with further manual testing based on your inputs.

**Comment by bitwiseman** [comment]
Okay, we now have an incremental that we can work with: [branch-api/2.5.9-rc884.9435363784d2](https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/branch-api/2.5.9-rc884.9435363784d2/).

**Comment by bitwiseman** [comment]
@addib 
Thanks for doing that!  



> * In declarative pipelines, `quietPeriod(<num>)` is respected for both PR creation and merge.

Yay!

> * The specified quiet period takes effect immediately on the very first run. However, removing it takes effect only after the next run.

This is expected/known behavior, similar to the way build parameters work in declarative.  It is due to a limitation in Jenkins. 

> * The global quiet period is respected if not specified in Jenkinsfile. If specified in Jenkinsfile, this takes precedence over the global settings.
> * Manual repository scans too make the build wait until the quiet period is elapsed.
> * PR builds triggered via PR comments (GitHub PR Comment Build Plugin 2.1) do not consider the quiet period. I guess this must be the expected behavior.

This is probably a bug in that plugin that should be filed and fixed separately.

> * The quiet period is not visible in multibranch project configuration GUI but is visible in the branch configuration GUI, though it remains uneditable.

This is a separate issue.  Please file it on the Jenkins JIRA with screenshots (if it hasn't already been filed).

**Comment by bitwiseman** [comment]
I'm trying this with some downstream plugins.

**Comment by hemp** [comment]
Found this in searching for a solution to multiple merges coming in at once on github.  Not a typical merge pattern for feature PRs...but with PRs made by dependabot doing a bunch of minor bumps on a regular cadence seems to be common.

Not sure how to help.  Would running an RC be useful?

**Comment by bitwiseman** [comment]
The failure in https://github.com/jenkinsci/workflow-multibranch-plugin/pull/99 should be fixed now. 
Moving on to try on bitbucket-branch-source and github-branch-source. 

@hemp 
Please go ahead and install the incremental linked above and report back anything you find.

**Comment by hemp** [comment]
We've had this live today and I reviewed our build history.  This is working great with a 30 second quiet period for a github org with a few hundred repos.  Merging multiple dependabot PRs (highs / criticals) can now be automated.  Thanks!  🎉

**Pull request merged**